src/legacy/ui/public/courier/search_source/search_source.js

```diff
@@ -312,7 +312,12 @@ export function SearchSourceProvider(Promise, Private, config) {
       }
 
       fetchSoon.fetchSearchRequests([req]);
-      return req.getCompletePromise();
+      return new Promise((resolve, reject) => {
+        req.getCompleteOrAbortedPromise().then(result => {
+          if (result instanceof Error) return reject(result);
+          resolve(result);
+        });
+      });
```

Comment:
This was annoying to debug. Basically what was happening in `searchSource.fetch()` is that it was returning this `req.getCompletePromise()`, but this promise never resolves or rejects when the request is aborted. So instead of returning it directly, I return a new promise that will reject with the aborted error if it's what resolves first.